services/QuillLMS/app/models/unit_activity.rb

```diff
@@ -124,6 +124,7 @@ def self.get_classroom_user_profile(classroom_id, user_id)
           COALESCE(cuas.locked, false) AS locked,
           COALESCE(cuas.pinned, false) AS pinned,
           MAX(acts.percentage) AS max_percentage,
+          SUM(CASE WHEN acts.state = 'finished' THEN 1 ELSE 0 END) > 0 AS finished,
```

Comment:
Not urgent but would probably be good to use a constant here (we may already have one?)